File: platforms/swift/examples/scene/scene/ambient.swift
import FragmentColor

let renderer = Renderer()
let target = try await renderer.createTextureTarget([64, 64])

let mesh = Mesh()
try mesh.addVertex(
    try Vertex.pbr([0.0, 0.5, 0.0]).set("uv0", [0.5, 1.0])
)

let scene = Scene()
// Warm dusk ambient — applies to every Material added below.
try scene.ambient([0.06, 0.04, 0.03])
try scene.add(Model(mesh, Material.pbr()))
try scene.add(Light.directional([0.3, -1.0, -0.4], [1.0, 0.95, 0.9]))

try renderer.render(scene, target)

File: platforms/kotlin/examples/scene/scene/ambient.kt
import org.fragmentcolor.*

val renderer = Renderer()
val target = renderer.createTextureTarget(64u, 64u)

val mesh = Mesh()
mesh.addVertex( Vertex.pbr(listOf(0.0f, 0.5f, 0.0f)).set("uv0", floatArrayOf(0.5f, 1.0f)), )

val scene = Scene()
// Warm dusk ambient — applies to every Material added below.
scene.ambient(listOf(0.06f, 0.04f, 0.03f))
scene.add(Model(mesh, Material.pbr()))
scene.add(Light.directional(listOf(0.3f, -1.0f, -0.4f), listOf(1.0f, 0.95f, 0.9f)))

renderer.render(scene, target)